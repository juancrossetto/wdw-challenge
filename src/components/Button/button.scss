.custom-button {
  border-radius: 8px !important;
  cursor: pointer;
}
