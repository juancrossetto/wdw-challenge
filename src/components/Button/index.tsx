import React, { FC } from 'react';
import { Button as BootstrapButton, ButtonProps as BootstrapButtonProps } from 'react-bootstrap';
import './button.scss';

interface ButtonProps {
  children?: JSX.Element | string;
}

const Button: FC<BootstrapButtonProps & ButtonProps> = ({ children, ...props }) => {
  return (
    <BootstrapButton {...props} className={`custom-button ${props.className}`} data-testid={'btn-test'}>
      {children}
    </BootstrapButton>
  );
};

export default Button;
