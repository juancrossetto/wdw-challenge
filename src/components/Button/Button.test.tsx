import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import Button from '.';

describe('Button component', () => {
  test('Button render correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Button with value render correctly', () => {
    const tree = renderer.create(<Button value={'some value'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Button onClick', () => {
    const mockCallback = jest.fn();
    const { getByTestId } = render(<Button value={'test'} onClick={mockCallback} />);
    const input = getByTestId('btn-test');
    fireEvent.click(input);
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });
});
