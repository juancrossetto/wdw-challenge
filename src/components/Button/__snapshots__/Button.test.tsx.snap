// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button render correctly 1`] = `
<button
  className="custom-button undefined btn btn-primary"
  data-testid="btn-test"
  disabled={false}
  type="button"
/>
`;

exports[`Button component Button with value render correctly 1`] = `
<button
  className="custom-button undefined btn btn-primary"
  data-testid="btn-test"
  disabled={false}
  type="button"
  value="some value"
/>
`;
